warrior/WarriorCore/Classes/argument_datatype_class.py

```diff
@@ -64,41 +64,44 @@ def convert_arg_to_datatype(self):
             self.datatype = tuple
         elif self.arg_name.startswith('dict_'):
             self.datatype = dict
+        elif self.arg_name.startswith('file_'):
+            self.datatype = file
+            tc_path = config_Utils.tc_path
+            fname = file_Utils.getAbsPath(self.arg_value, tc_path)
+            try:
+                self.arg_value = file(fname)
+            except IOError:
+                print_warning("given file {} does not exist, please check, it "
+                              "should be relative to testcase path {}".format(fname, tc_path))
 
         else:
-            #print_info("User has not specified any data type,
-            #input argument will be treated as string (default)")
             return self.arg_value
-#         if self.datatype is not None:
-#             convert_msg = "Input argument {0} will be converted to a {1}".format(self.arg_name,
-#                                                                                  self.datatype)
-#             print_info(convert_msg)
-#
+        if self.datatype is not None:
+            convert_msg = "Input argument {0} will be converted to a {1}".format(
+                                                        self.arg_name, self.datatype)
+            print_info(convert_msg)
+
         result = self.convert_string_to_datatype()
         return result
 
     def convert_string_to_datatype(self):
         """Converts an input string to a python datatype """
 
-        err_msg = "User input argument value {0} does"\
-        "not match python syntax for '{1}'".format(self.arg_value, self.datatype)
-        info_msg = "Warrior FW will handle user input argument value as string (default)"
+        err_msg = ("\nUser input argument value '{0}' does not match python "
+                   "syntax for '{1}'").format(self.arg_value, self.datatype)
+        info_msg = "Warrior FW will handle user input argument value as string (default)\n"
+        result = self.arg_value
+
         try:
-            result = ast.literal_eval(self.arg_value)
+            if self.datatype is not file:
+                result = ast.literal_eval(self.arg_value)
```

Comment:
if datatype is file
should result be the content of file?

Reply:
No, it should be file object